### PR TITLE
[frdb] Add support for env connection variable.

### PR DESCRIPTION
**Pull request by awdavies**
Similar to the flutter driver, now the fuchsia_remote_debug_protocol can
connect via an environment variable. This can also be used to point to
the SSH config location.

This makes it so that tests written using the FRDB do not have to write
extra information about where their device is on the network.

### CODE REVIEW

File: packages/fuchsia_remote_debug_protocol/lib/src/fuchsia_remote_connection.dart [hunk 1]
```diff
@@ -13,9 +13,11 @@ import 'common/network.dart';
 import 'dart/dart_vm.dart';
 import 'runners/ssh_command_runner.dart';
 
-final String _ipv4Loopback = InternetAddress.LOOPBACK_IP_V4.address; // ignore: deprecated_member_use
+final String _ipv4Loopback =
+    InternetAddress.LOOPBACK_IP_V4.address; // ignore: deprecated_member_use
```
Comment (tvolkert) [inline packages/fuchsia_remote_debug_protocol/lib/src/fuchsia_remote_connection.dart]: Why not use the non-deprecated value?

File: packages/fuchsia_remote_debug_protocol/lib/src/fuchsia_remote_connection.dart [hunk 2]
```diff
@@ -47,6 +49,21 @@ void restoreFuchsiaPortForwardingFunction() {
...
+class FuchsiaRemoteConnectionError extends Error {
+  /// Basic constructor outlining the reason for the failure in `message`.
+  FuchsiaRemoteConnectionError(this.message);
+
+  /// The reason for the failure.
+  final String message;
+
+  @override
+  String toString() {
+    return '$FuchsiaRemoteConnectionError: $message\n${super.stackTrace}';
```
Comment (tvolkert) [inline packages/fuchsia_remote_debug_protocol/lib/src/fuchsia_remote_connection.dart]: The stack trace should show up if it's uncaught - you shouldn't need to add it to `toString()`.  If it's caught and being logged, then the place where it's caught should record the stack trace.

File: packages/fuchsia_remote_debug_protocol/lib/src/fuchsia_remote_connection.dart (hunk 1 above)
Comment (awdavies) [reply packages/fuchsia_remote_debug_protocol/lib/src/fuchsia_remote_connection.dart]: This is just from the auto-formatter.  I can change it, though.

**Review by tvolkert — APPROVED** [review]

File: packages/fuchsia_remote_debug_protocol/lib/src/fuchsia_remote_connection.dart (hunk 2 above)
Comment (awdavies) [reply packages/fuchsia_remote_debug_protocol/lib/src/fuchsia_remote_connection.dart]: Fixed.

**Comment by Hixie** [comment]
Please don't land patches when one of the pre-commit tests is red, even if it's a harmless error or if the PR can "obviously" not be wrong, because it makes trying to track down the problem when the PR subsequently makes the tree red much harder. If it's a flake, rerun the test (and file a bug to fix the flake, and fix the flake). If it's a configuration problem, please let me know and wait before landing the PR.

**Comment by Hixie** [comment]
(no test for this PR?)

**Comment by awdavies** [comment]
@Hixie How do we force the tests to re-run?

**Comment by Hixie** [comment]
Depends on the precise system, but generally click "Details" to see the logs, then on that page log in, and that causes some button to appear for restarting the system.

**Comment by awdavies** [comment]
Thanks!  Also totally spaced on testing.  I'll add a PR for that :+1:

**Pull request merged**